server/api/badge.go

```diff
@@ -72,7 +73,14 @@ func GetBadge(c *gin.Context) {
 		branch = repo.Branch
 	}
 
-	pipeline, err := _store.GetPipelineBadge(repo, branch)
+	// Events to lookup in the pipe, set multiple separated by comma
+	// If none given, fallback to former default "push"
+	events := c.Query("events")
+	if len(events) == 0 {
+		events = string(model.EventPush)
+	}
+
+	pipeline, err := _store.GetPipelineBadge(repo, branch, strings.Split(events, ","))
```

Comment:
This should be validated and if there is at least one event that does not exist a 400 error should be returned.

Should be possible to do like this:
```suggestion
	// Events to lookup, multiple separated by comma
	var events []*model.WebhookEvent
	eventsQuery := c.Query("events")

	// If none given, fallback to default "push"
	if len(events) == 0 {
		events = []*model.WebhookEvent{model.EventPush)}
	} else {
		for _, event := range strings.Split(events, ",") {
			e := model.WebhookEvent(event)
			if err := e.Validate(); err != nil {
				c.AbortWithError(http.StatusBadRequest, err)
				return
			}
		}
	}

	pipeline, err := _store.GetPipelineBadge(repo, branch, events)
```

And then also change the `GetPipelineBadge` signature to use `[]*model.WebhookEvent`.

Reply:
If I understood correctly, it wasn't possible to use a reference ([]*model.WebhookEvent) as the WebhookEvents are consts and cannot be used by addresses. So I changed it to []model.WebhookEvent.